gcp_airflow_foundations/source_class/source.py

```diff
@@ -54,60 +45,129 @@ def build_dags(self):
+                table_config = templated_table_config
+                table_config.table_name = table_name
+            else:
+                table_config = self.config.tables[i]
+
             logging.info(f"table_default_task_args {table_default_task_args}")
 
+            start_date = table_default_task_args["start_date"]
+            
             kwargs = data_source.dag_args if data_source.dag_args else {}
-
+            
             with DAG(
                 dag_id=f"{data_source.name}.{table_config.table_name}",
                 description=f"{data_source.name} to BigQuery load for {table_config.table_name}",
                 schedule_interval=data_source.ingest_schedule,
                 default_args=table_default_task_args,
-                catchup=data_source.catchup,
                 render_template_as_native_obj=True,
-                **kwargs,
-            ) as dag:
+                **kwargs
+            ) as dag:    
 
-                load_to_bq_landing = self.get_bq_ingestion_task(dag, table_config)
+                self.set_task_dependencies(dag, table_config, data_source, "schema_parsing")
+                dags.append(dag)
 
-                self.get_datastore_ingestion_task(
-                    dag, load_to_bq_landing, data_source, table_config
-                )
+        extra_dags=  self.get_extra_dags()
 
-                dags.append(dag)
+        if extra_dags is not None and not extra_dags == []:
+            dags = dags + self.get_extra_dags()
 
-        dags = dags + self.get_extra_dags()
         return dags
 
+    def build_dags_source_level(self):
+        """
+        Build one DAG per source for regex matching or full source ingestion 
+        """
+        data_source = self.config.source
+
+        logging.info(f"Building DAG for {data_source.name}")
+
+        dags = []
+        tables = self.get_source_tables_to_ingest()
+
+        # one template table config is filled out for the entire source
+        templated_table_config = self.config.tables[0]
+        table_default_task_args = self.default_task_args_for_table(
+            self.config, templated_table_config
+        )
+        logging.info(f"table_default_task_args {table_default_task_args}")
+
+        start_date = table_default_task_args["start_date"]
+        
+        kwargs = data_source.dag_args if data_source.dag_args else {}
+        
+        with DAG(
+            dag_id=f"{data_source.name}.{data_source.full_ingestion_options.ingestion_name}",
+            description=f"{data_source.name} to BigQuery load",
+            schedule_interval=data_source.ingest_schedule,
+            default_args=table_default_task_args,
+            render_template_as_native_obj=True,
+            **kwargs
+        ) as dag:    
+            
+            for table_name in tables:
+                table_config = templated_table_config
+                table_config.table_name = table_name
+
+                schema_parsing_task_id = "schema_parsing"
```

Comment:
If it is a constant, move it to the top level and capitalize 

SCHEMA_PARSING_TASK_ID = "schema_parsing"